pkg/operator/lockserver/params.go

```diff
@@ -33,15 +33,15 @@ const (
 
 // GlobalConnectionParams returns the Vitess connection parameters for a
 // VitessCluster's global lockserver.
-func GlobalConnectionParams(vt *planetscalev2.VitessCluster) *planetscalev2.VitessLockserverParams {
+func GlobalConnectionParams(lockSpec *planetscalev2.LockserverSpec, name string) *planetscalev2.VitessLockserverParams {
```

Comment:
Is this pointer necessary?

Reply:
I might be wrong, but I believe without a pointer then we have copy semantics, meaning the struct is copied when the function is called. The pointer eliminates unnecessary struct copying. Perhaps though with such a small struct it doesn't save us much in terms of memory and might be slower due to heap allocation. @enisoc, thoughts on this?I might be wrong, but I believe without a pointer then we have copy semantics, meaning the struct is copied when the function is called. The pointer eliminates unnecessary struct copying. Perhaps though with such a small struct it doesn't save us much in terms of memory and might be slower due to heap allocation. @enisoc, thoughts on this?

Reply:
I prefer to pass k8s API structs as pointers in most cases, if only because these structs very often contain pointers themselves, which means a shallow copy only sets you up to mistakenly believe you have a copy when you actually don't. It's also a habit because the scalability team in k8s hounded us to avoid copies in the core controllers, since they identified the copies and extra GC load as being a problem for controllers that handle lots of objects.

Reply:
Sounds like this was the right approach then. I'll mark this resolved for now and go ahead and merge.